test/extended/baremetal/hosts.go

```diff
@@ -187,3 +187,72 @@ var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] Baremetal platfor
 		o.Expect(found).To(o.BeFalse())
 	})
 })
+
+var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] A baremetal deployment without a provisioning network should", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc = exutil.NewCLI("baremetal")
+	)
+
+	g.It("show the Provisioning Network as 'Disabled'", func() {
+		skipIfNotBaremetal(oc)
+
+		dc := oc.AdminDynamicClient()
+
+		skipIfProvisioningNetworkSet(dc)
+
+		o.Expect(getProvisioningNetwork(dc)).To((o.BeEquivalentTo("Disabled")))
+
+		g.By("Not allow setting the ProvisioningNetwork to 'Managed' with invalid values")
+		invalidProvisioningNetworkCIDR := "172.22.0.0/33"
+		provisioningClient := provisioningClient(dc)
+
+		provisionings, err := provisioningClient.List(context.Background(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(provisionings.Items).ToNot(o.BeEmpty())
+
+		provisioning := provisionings.Items[0]
+
+		err = unstructured.SetNestedField(provisioning.Object, invalidProvisioningNetworkCIDR, "spec", "provisioningNetworkCIDR")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		_, err = provisioningClient.Update(context.Background(), &provisioning, metav1.UpdateOptions{})
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(err.Error()).To(o.ContainSubstring("could not parse provisioningNetworkCIDR"))
+
+		o.Expect(getProvisioningNetwork(dc)).To((o.BeEquivalentTo("Disabled")))
+	})
+
+	g.It("allow setting the ProvisioningNetwork to 'Managed' with valid settings", func() {
+		skipIfNotBaremetal(oc)
+
+		dc := oc.AdminDynamicClient()
+
+		skipIfProvisioningNetworkSet(dc)
+
+		validProvisioningNetworkCIDR := "172.22.0.0/24"
+		validProvisioningIP := "172.22.0.3"
+
+		provisioningClient := provisioningClient(dc)
+
+		provisionings, err := provisioningClient.List(context.Background(), metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(provisionings.Items).ToNot(o.BeEmpty())
+
+		provisioning := provisionings.Items[0]
+
+		err = unstructured.SetNestedField(provisioning.Object, validProvisioningNetworkCIDR, "spec", "provisioningNetworkCIDR")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = unstructured.SetNestedField(provisioning.Object, validProvisioningIP, "spec", "provisioningIP")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = unstructured.SetNestedField(provisioning.Object, "Managed", "spec", "provisioningNetwork")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		_, err = provisioningClient.Update(context.Background(), &provisioning, metav1.UpdateOptions{})
```

Comment:
We are changing `provisioningIP` and `provisioningNetwork` but we are not reverting back after test is completed. Is it safe for the other tests?

Reply:
I guess we could move the provisioningNetwork back to Disabled ?

Reply:
Yes, I think so. Otherwise, other tests might be affected.